spec/install/deploy_spec.rb

```diff
@@ -295,6 +295,106 @@
       expect(out).not_to include("You have deleted from the Gemfile")
     end
 
+    it "allows to replace a host with the same host with credentials when allow_deployment_source_credential_changes is true" do
```

Comment:
`when allow_deployment_source_credential_changes is true` this should be in a context block